app/models/nomis/sentence_detail.rb

```diff
@@ -29,13 +29,27 @@ def conditional_release_date
 
     def earliest_release_date
       dates = [
-          release_date,
+          automatic_release_date,
+          conditional_release_date,
+          home_detention_curfew_actual_date,
+          home_detention_curfew_eligibility_date,
           parole_eligibility_date,
           tariff_date
       ].compact
       return nil if dates.empty?
 
-      dates.min.to_date
+      past_dates = []
+      future_dates = []
+
+      dates.each do |date|
+        date >= Time.zone.today ? future_dates << date : past_dates << date
```

Comment:
I found this a little hard to parse - I'll happily use ternary operators for assignment:
```ruby
some_variable = condition ? first_possibility : second_possibility
```

but I'll rarely use it for operations unless they're all methods
```ruby
# happy with this
condition ? do_thing : do_second_thing

# starting to lose track!
some + maths ? comparisons > thing : otherwise << thing_two
```

My first thought is to split this up a little (not sure if this is exactly right, but feels closer):
```ruby
date_has_passed = date >= Time.zone.today
dates_to_compare = date_has_passed ? future_dates : past_dates
dates_to_compare << date
```

Or maybe just a straight `if` is cleaner?
```ruby
if date >= Time.zone.today 
  future_dates << date
else
  past_dates << date
end
```

Reply:
Refactored this section of the method; hopefully making it a little easier to read